Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs

```diff
@@ -0,0 +1,95 @@
+﻿using EFCoreAssignment.Data.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace EFCoreAssignment.Data.Services
+{
+    public class ProductService : IProductService
+    {
+        private readonly AppDbContext _appDbContext;
+
+        public ProductService(AppDbContext appDbContext)
+        {
+            _appDbContext = appDbContext;
+        }
+
+        public async Task<List<ProductDto>> GetProducts()
+        {
+            var products = await _appDbContext.Products.ToListAsync();
+            if (products == null)
+            {
+                return null;
+            }
+
+            var productsDto = products.Select(p => new ProductDto(p.Id, p.Name, p.ShopId)).ToList();
+
+            return productsDto;
+        }
+
+        public async Task<ProductDto> GetProduct(int id)
+        {
+            var product = await _appDbContext.Products.FirstOrDefaultAsync(p => p.Id == id);
+            if (product == null)
+            {
+                return null;
+            }
+
+            var productDto = new ProductDto(product.Id, product.Name, product.ShopId);
+
+            return productDto;
+        }
+
+        public async Task<int> CreateProduct(CreateProductDto productForCreation)
+        {
+            var product = new Product
+            {
+                Name = productForCreation.Name,
+                ShopId = productForCreation.ShopId
+            };
+
+            _appDbContext.Products.Add(product);
+            await _appDbContext.SaveChangesAsync();
+
+            return product.Id;
+        }
+
+        public async Task UpdateProduct(UpdateProductDto productForUpdate)
+        {
+            var product = await _appDbContext.Products.FirstOrDefaultAsync(p => p.Id == productForUpdate.Id);
+            if (product == null)
+            {
+                throw new ArgumentException("Product not found.");
```

Comment:
You can handle the checking of the product in the controller before calling the UpdateProduct() in service

Reply:
Updated code. Thanks!